### PR TITLE
[G-API] Support RMat for PlaidML backend

### PR DESCRIPTION
**Pull request by TolyaTalamanov**
### Pull Request Readiness Checklist

See details at https://github.com/opencv/opencv/wiki/How_to_contribute#making-a-good-pull-request

- [ ] I agree to contribute to the project under Apache 2 License.
- [ ] To the best of my knowledge, the proposed patch is not based on a code under GPL or other license that is incompatible with OpenCV
- [ ] The PR is proposed to proper branch
- [ ] There is reference to original bug report and related work
- [ ] There is accuracy test, performance test and test data in opencv_extra repository, if applicable
      Patch to opencv_extra has the same branch name.
- [ ] The feature is well documented and sample code can be built with the project CMake

### Build configuration
```
Xforce_builders_only=linux,docs
force_builders=Custom,Custom Win,Custom Mac
build_gapi_standalone:Linux x64=ade-0.1.1f
build_gapi_standalone:Win64=ade-0.1.1f
build_gapi_standalone:Mac=ade-0.1.1f
build_gapi_standalone:Linux x64 Debug=ade-0.1.1f

Xbuild_image:Custom=centos:7
Xbuildworker:Custom=linux-1
build_gapi_standalone:Custom=ade-0.1.1f

build_image:Custom Win=openvino-2021.1.0
build_image:Custom Mac=openvino-2021.2.0

test_modules:Custom=gapi,python2,python3,java
test_modules:Custom Win=gapi,python2,python3,java
test_modules:Custom Mac=gapi,python2,python3,java

buildworker:Custom=linux-1
# disabled due high memory usage: test_opencl:Custom=ON
test_opencl:Custom=OFF
Xtest_bigdata:Custom=1
Xtest_filter:Custom=*

Xforce_builders=Custom
Xbuildworker:Custom=linux-1
build_image:Custom=plaidml2
Xtest_modules:Custom=gapi
test_filter:Custom=*ML*
```

### CODE REVIEW

File: modules/gapi/src/backends/plaidml/gplaidmlbackend.cpp [hunk 1]
```diff
@@ -215,10 +212,12 @@ void cv::gimpl::GPlaidMLExecutable::bindInArg(const RcDesc &rc, const GRunArg  &
...
-        case GRunArg::index_of<cv::Mat>() :
+        case GRunArg::index_of<cv::RMat>():
         {
-            auto& arg_mat = util::get<cv::Mat>(arg);
-            binder_->input(it->second).copy_from(arg_mat.data);
+            auto& rmat = cv::util::get<cv::RMat>(arg);
+            auto  view = rmat.access(cv::RMat::Access::R);
+            auto  mat  = cv::gimpl::asMat(view);
+            binder_->input(it->second).copy_from(mat.data);
         }
```
Comment (dmatveev) [inline modules/gapi/src/backends/plaidml/gplaidmlbackend.cpp:221]: @rgarnov shouldn't this happen by default? I thought it is

**Review by dmatveev — APPROVED** [review]
@rgarnov do we have any other backends with an issue like this or _this_ backend just didn't use magazine which "hides" this new detail?

**Comment by TolyaTalamanov** [comment]
@alalek 

According log PlaidML wasn't found
```
[----------] 1 test from GAPI_PlaidML_Pipelines
[ RUN      ] GAPI_PlaidML_Pipelines.ThrowIfPlaidMLNotFound
[       OK ] GAPI_PlaidML_Pipelines.ThrowIfPlaidMLNotFound (0 ms)
[----------] 1 test from GAPI_PlaidML_Pipelines (0 ms total)
```

doesn't it work anymore ? 
```
force_builders=Custom
buildworker:Custom=linux-1
build_image:Custom=plaidml2
test_modules:Custom=gapi
test_filter:Custom=*ML*
```

**Comment by alalek** [comment]
> `build_image:Custom=plaidml2`

This would work if we remove `build_image:Custom=ubuntu-openvino-2021.1.0:20.04` above (or comment out).

**Comment by TolyaTalamanov** [comment]
Now, it's OK, thanks!
```
[==========] Running 3 tests from 1 test case.
[----------] Global test environment set-up.
[----------] 3 tests from GAPI_PlaidML_Pipelines
[ RUN      ] GAPI_PlaidML_Pipelines.SimpleArithmetic
[       OK ] GAPI_PlaidML_Pipelines.SimpleArithmetic (137 ms)
[ RUN      ] GAPI_PlaidML_Pipelines.TwoInputOperations
[       OK ] GAPI_PlaidML_Pipelines.TwoInputOperations (82 ms)
[ RUN      ] GAPI_PlaidML_Pipelines.TwoOutputOperations
[       OK ] GAPI_PlaidML_Pipelines.TwoOutputOperations (87 ms)
[----------] 3 tests from GAPI_PlaidML_Pipelines (306 ms total)

[----------] Global test environment tear-down
[==========] 3 tests from 1 test case ran. (306 ms total)
[  PASSED  ] 3 tests.
```

File: modules/gapi/src/backends/plaidml/gplaidmlbackend.cpp (hunk 1 above)
Comment (rgarnov) [reply modules/gapi/src/backends/plaidml/gplaidmlbackend.cpp:221]: It's done in default magazine::bindInArg. Default magazine is not used in plaidml backend and I think it's not worth to bring it here. So current changes are good

**Review by rgarnov — APPROVED** [review]

**Pull request merged**